### PR TITLE
Fix to add place on tap place list suggestion

### PR DESCRIPTION
**Pull request by kaushiksaliya**
Place list suggestion.

- on tap place list suggestion add place.
- Add place name text field and show address according to position on locate on map screen.
- Minor fixes on action button.

[Fix_suggestion_place_click.webm](https://github.com/user-attachments/assets/d834ec49-deeb-42de-8b0e-5693ce7a2494)

### CODE REVIEW

**Review by cp-radhika-s — APPROVED** [review]

File: app/lib/ui/flow/geofence/add/locate/locate_on_map_screen.dart [hunk 1]
```diff
@@ -73,54 +89,102 @@ class _LocateOnMapViewState extends ConsumerState<LocateOnMapScreen> {
...
+    return Column(
+      children: [
+        AppTextField(
+          controller: TextEditingController(text: widget.placesName),
+          enabled: false,
+          style: AppTextStyle.subtitle2
+              .copyWith(color: context.colorScheme.textPrimary),
+          onTapOutside: (event) {
+            FocusManager.instance.primaryFocus?.unfocus();
+          },
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/geofence/add/locate/locate_on_map_screen.dart:154]: this event already added in AppTextField, no need to add here

File: app/lib/ui/flow/geofence/places/places_list_screen.dart [hunk 2]
```diff
@@ -80,7 +80,9 @@ class _PlacesViewState extends ConsumerState<PlacesListScreen> {
                 placeName: item,
                 icon: _getPlacesIcon(item),
                 isSuggestion: true,
-                onTap: () {},
+                onTap: () {
+                  onSuggestionItemTap(item);
+                },
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/geofence/places/places_list_screen.dart:85]: you can make code structure like for single line `onTap: () => onSuggestionItemTap(item),`

**Comment by cp-ishita-g** [comment]
not able to move to next screen on tap on item 


https://github.com/user-attachments/assets/8fb05a54-8f47-486b-b98f-28e7c58117d0

File: app/lib/ui/flow/geofence/add/addnew/add_new_place_screen.dart [hunk 3]
```diff
@@ -110,7 +110,7 @@ class _AddNewPlaceViewState extends ConsumerState<AddNewPlaceScreen> {
   Widget _locateOnMapView() {
     return OnTapScale(
       onTap: () {
-        AppRoute.locateOnMapScreen(widget.spaceId).push(context);
+        AppRoute.locateOnMapScreen(spaceId:widget.spaceId).push(context);
```
Comment (cp-ishita-g) [inline app/lib/ui/flow/geofence/add/addnew/add_new_place_screen.dart:113]: add space after `spaceId:`

**Comment by kaushiksaliya** [comment]
> app progress indictor should be centre aligned
> 
>  Simulator.Screen.Recording.-.iPhone.15.Plus.-.2024-07-25.at.17.44.15.mp4



> not able to move to next screen on tap on item
> 
>  Simulator.Screen.Recording.-.iPhone.15.Plus.-.2024-07-25.at.17.42.08.mp4

@cp-ishita-g update branch because of i am able to tap and also showing next screen.

**Comment by cp-ishita-g** [comment]
fix this UI issue

Simulator Screenshot - iPhone 15 Pro - 2024-07-25 at 18 15 52

**Comment by cp-ishita-g** [comment]
fix font color issue
Simulator Screenshot - iPhone 15 Pro - 2024-07-25 at 18 18 57

**Comment by cp-ishita-g** [comment]
use native alert pop for iOS and android as we have used in another places

Simulator Screenshot - iPhone 15 Pro - 2024-07-25 at 18 21 09

**Comment by cp-ishita-g** [comment]
change switch on and off color to our primary color

Simulator Screenshot - iPhone 15 Pro - 2024-07-25 at 18 24 18

**Comment by cp-ishita-g** [comment]
delete space button should be bottom sticky 

https://github.com/user-attachments/assets/f9631dc5-c210-4a62-8a4b-794c67e97f8a

**Comment by cp-ishita-g** [comment]
you forgot to add some suggestion on add place screen according to design

https://www.figma.com/design/M8j8p90gMoHYlxZqVkRxvV/YourSpace?node-id=390-3194&t=bYkuM0oFSkMJ8zlp-0

**Comment by cp-ishita-g** [comment]
according to design, there is no bottom zoom in zoom out support for map

Simulator Screenshot - iPhone 15 Pro - 2024-07-25 at 18 33 03

**Review by cp-ishita-g — APPROVED** [review]

**Pull request merged**